src/storage/page_buffer.c

```diff
@@ -7773,31 +7750,40 @@ pgbuf_claim_bcb_for_fix (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_
 	  (void) pgbuf_unlock_page (thread_p, hash_anchor, vpid, true);
 
 #if defined(ENABLE_SYSTEMTAP)
+	  bool monitored = false;
+	  QUERY_ID query_id = NULL_QUERY_ID;
+
+	  query_id = qmgr_get_current_query_id (thread_p);
+	  if (query_id != NULL_QUERY_ID)
+	    {
+	      monitored = true;
+	      CUBRID_IO_READ_START (query_id);
+	    }
+
 	  if (monitored == true)
 	    {
 	      CUBRID_IO_READ_END (query_id, IO_PAGESIZE, 1);
 	    }
 #endif /* ENABLE_SYSTEMTAP */
-
 	  PGBUF_BCB_CHECK_MUTEX_LEAKS ();
-	  return NULL;
 	}
 
-#if defined(SERVER_MODE)
-	// *INDENT-OFF*
-	if (get_server_type () == SERVER_TYPE_TRANSACTION && ats_Gl.is_page_server_connected ()) 
-	  {
-	    auto data_page = ats_Gl.get_data_page_broker ().wait_for_page (*vpid);
-	
-	    char buf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
-	    FILEIO_PAGE *io_page = (FILEIO_PAGE *) PTR_ALIGN (buf, MAX_ALIGNMENT);
-	    std::memcpy (io_page, data_page->c_str (), db_io_page_size ());
-
-	    assert (bufptr->iopage_buffer->iopage.prv == io_page->prv);
-    	  }
-	// *INDENT-ON*
-#endif // SERVER_MODE
+#if defined(ENABLE_SYSTEMTAP)
+      bool monitored = false;
+      QUERY_ID query_id = NULL_QUERY_ID;
 
+      query_id = qmgr_get_current_query_id (thread_p);
+      if (query_id != NULL_QUERY_ID)
+	{
+	  monitored = true;
+	  CUBRID_IO_READ_START (query_id);
+	}
+
+      if (monitored == true)
+	{
+	  CUBRID_IO_READ_END (query_id, IO_PAGESIZE, 1);
+	}
+#endif /* ENABLE_SYSTEMTAP */
```

Comment:
Remove this. Find the explanation above how CUBRID_IO_READ_START/CUBRID_IO_READ_END should be used.